src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -242,10 +242,13 @@ internal sealed override IEnumerable<Cci.INestedTypeDefinition> GetSynthesizedNe
                                         // NOTE: Dev11 does not add synthesized static constructors to this map,
                                         //       but adds synthesized instance constructors, Roslyn adds both
                                         var method = (MethodSymbol)member;
-                                        if (!method.IsDefaultValueTypeConstructor())
+                                        if (method.IsDefaultValueTypeConstructor() ||
+                                            method.IsPartialMethod() && (object)method.PartialImplementationPart == null)
```

Comment:
Current condition matches condition in GetMethodsToEmit. If we decide to change it, both places should be changed.